packages/yew-router/src/components/link.rs

```diff
@@ -49,20 +54,50 @@ where
         let navigator = navigator.clone();
         let to = to.clone();
         let query = query.clone();
+        let target = target.clone();
 
         Callback::from(move |e: MouseEvent| {
             e.prevent_default();
 
-            match query {
-                None => {
-                    navigator.push(&to);
-                }
-                Some(ref data) => {
-                    navigator
-                        .push_with_query(&to, data)
-                        .expect_throw("failed push history with query");
+            match target {
+                Some(ref target) => {
```

Comment:
When target is `Some(_)`, I feel it's better to pass `target` to the anchor element and don't call `e.prevent_default()` here. The link will be handled by the browser.